src/site/shared/circuitdesigner/impl/CircuitDesigner.ts

```diff
@@ -39,6 +43,17 @@ export class CircuitDesignerImpl<CircuitT extends Circuit> implements CircuitDes
 
         // Setup tool manager
         this.inputAdapter.subscribe((ev) => this.toolManager.onEvent(ev, this));
+
+        // Attach tool renderers
+        if (renderers) {
+            this.circuit.addRenderCallback(({ renderer, options, circuit }) => {
+                renderers.forEach((toolRenderer) => {
+                    const curTool = this.toolManager.curTool;
+                    if (toolRenderer.isActive(curTool))
```

Comment:
Could you keep track of which tool renderer(s) are associated with each tool on construction and avoid this`isActive` pattern.  Making it something like:

```typescript
this.circuit.addRenderCallback(({ renderer, options, circuit }) => {
    this.toolManager.curToolRenderer.render({ renderer, options, circuit, this.toolManager.curTool, input: this.inputAdapter.state });
});
```
And then encapsulate each tool renderer with its associated tool once on construction for static type safety.

Reply:
I don't think so -- like the Rotate Tool (as seen in future PRs) will need to be active while rotating but also for the DefaultTool when something is selected to show the rotation circle, and theoretically other tool renderers could also effect the DefaultTool

Reply:
I see.  I think this is a limitation of the current Tools state machine model, then.  A model that might fit this Rotate Tool use-case could be: 
- Each tool has one state of; `Active`, `Pending`, and `Inactive`.
- At any given time there is one `Active` tool.
- All tools that are accessible from the current state are `Pending`.
- All other tools are `Inactive`

Each tool renderer is run every frame, deciding what, if anything, should be rendered based on the state of its associated tool.  For example, when a component is selected, the rotate tool is `Pending`, and so the rotate tool renderer would render the tool for the `Pending` state.

This is an over-simplification, but do you understand what I'm getting at?

Reply:
Going to handle this in a following PR as not to clutter this one